src/v/cluster/partition_balancer_planner.cc

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cluster/partition_balancer_planner.h"
+
+#include "cluster/health_monitor_types.h"
+#include "cluster/scheduling/constraints.h"
+#include "cluster/scheduling/types.h"
+#include "cluster/types.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+
+#include <optional>
+#include <utility>
+
+namespace cluster {
+
+partition_balancer_planner::partition_balancer_planner(
+  planner_config config,
+  topic_table& topic_table,
+  partition_allocator& partition_allocator)
+  : _config(config)
+  , _topic_table(topic_table)
+  , _partition_allocator(partition_allocator) {}
+
+void partition_balancer_planner::
+  init_ntp_sizes_and_node_reports_from_health_report(
+    const cluster_health_report& health_report,
+    reallocation_request_state& rrs) {
+    for (const auto& node_report : health_report.node_reports) {
+        rrs.node_reports[node_report.id] = node_report;
+        for (const auto& tp_ns : node_report.topics) {
+            for (const auto& partition : tp_ns.partitions) {
+                rrs.ntp_sizes[model::ntp(
```

Comment:
this comment is still relevant I think https://github.com/redpanda-data/redpanda/pull/5142#discussion_r906105034